go/arrow/type_traits_float16.go

```diff
@@ -0,0 +1,95 @@
+package arrow
+
+import (
+	"encoding/binary"
+	"math"
+	"reflect"
+	"unsafe"
+)
+
+// Float16 traits
+
+type Float16 uint16
+
+// https://en.wikipedia.org/wiki/Half-precision_floating-point_format
```

Comment:
please add a proper documentation string.
keeping the wikipedia link is good, but having a bit of plain english is great :)